### PR TITLE
Use byRoot RPC for non-finalized DataColumnSidecars 

### PR DESCRIPTION
**Pull request by zilm13**
## PR Description
We could stuck using only byRange on non-finalized chain fragments, so added usage of byRoot.

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: infrastructure/async/src/test/java/tech/pegasys/teku/infrastructure/async/stream/AsyncStreamTest.java [hunk 1]
```diff
@@ -237,4 +237,18 @@ void testConcurrentExceptionHasUsefulWrap() throws Exception {
...
+    assertThat(
+            AsyncStream.of(0, 1, 2)
+                .takeUntil(i -> i == 0, false)
+                .merge(AsyncStream.of(3))
+                .toList()
+                .join())
+        .contains(3);
+    assertThat(AsyncStream.of(0, 1, 2).merge(AsyncStream.of(3)).toList().join())
+        .contains(0, 1, 2, 3);
+    assertThat(AsyncStream.of(0, 1, 2).merge(AsyncStream.of()).toList().join()).contains(0, 1, 2);
```
Comment (mehdi-aouadi) [inline infrastructure/async/src/test/java/tech/pegasys/teku/infrastructure/async/stream/AsyncStreamTest.java]: I'd add a test with duplicate elements

**Review by mehdi-aouadi — APPROVED** [review]
LGTM
Just a nit for an additional test
